File: backend/src/main/java/com/candleshop/backend/model/Product.java
package com.candleshop.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(length = 1000)
    private String description;

    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private CandleCategory category;

    private String scent;

    @Column(name = "burn_time")
    private String burnTime;

    private String size;

    private Double weight;

    @Column(name = "stock_status", nullable = false)
    private String stockStatus;

    private Double rating;

    @Column(name = "review_count")
    private Integer reviewCount;

    @Column(name = "is_featured")
    private boolean featured;

    @Column(name = "is_new_arrival")
    private boolean newArrival;

    @Column(name = "is_best_seller")
    private boolean bestSeller;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
} 

File: frontend/src/app/models/product.ts
export interface Product {
  id: number;
  name: string;
  price: number;
  description: string;
  imageUrl: string;
  category?: Category | number;  // Can be either a Category object or a category ID
  categoryId?: number;           // For API responses that include categoryId directly
  categoryName?: string;         // For API responses that include categoryName directly
  scent?: string;
  burnTime?: string;
  size?: string;
  weight?: string | number;
  inStock?: boolean;
  stockStatus?: string;          // For API responses that use stockStatus enum
  rating?: number;
  reviewCount?: number;
  featured?: boolean;
  newArrival?: boolean;
  bestSeller?: boolean;
  salesCount?: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface Category {
  id: number;
  name: string;
  description?: string;
  imageUrl?: string;
  slug?: string;
  productCount?: number;
}

export enum CandleCategory {
  SCENTED = 'Scented',
  UNSCENTED = 'Unscented',
  SOY = 'Soy',
  BEESWAX = 'Beeswax',
  FLOATING = 'Floating',
  PILLAR = 'Pillar',
  CONTAINER = 'Container',
  TEALIGHT = 'Tea Light',
  VOTIVE = 'Votive'
}
